.github/workflows/CD.yml

```diff
@@ -639,9 +647,46 @@ jobs:
         env:
           AWS_SDK_LOAD_CONFIG: true
 
+  clean-ssm-package:
+    runs-on: ubuntu-latest
+    if: ${{ always() }}
+    needs: [ s3-release-validation-1, s3-release-validation-2, s3-release-validation-3, release-validation-ecs, release-validation-eks, release-checking ]
+    steps:
+      #Delete ssm package and binaries/artifacts which is created on s3 bucket for validation test regardless of these two scenarios:
+      #-if the validation test failed, then we will delete the SSM package since we don't want to roll out the unstable version
+      #-if the validation successes, then we will trigger the release SSM package workflow to replace all of this with the appropriate region
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.RELEASE_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.RELEASE_SECRET }}
+          aws-region: us-west-2
+
+      - name: Restore cached packages
```

Comment:
What do you need the cached packages for?

Reply:
Mostly the cached packages are for checking if the SSM release is a success or not. However, add a condition and only trigger when```release-to-s3``` job success.